### DISCUSSION TITLE
When the value is 0, there is a problem with -0.0

### DISCUSSION THREAD

**Discussion by jskorlol** [livewire/livewire#3017] [7. Bugs (archived)] [answered]
### Description

When the public variable contains 0,
This value is -0.0; If changed to, an error occurs.

### Exact steps to reproduce

1. Click the test() button, and set the weight value to -0.0; To change it.
2. If you click the test1() button, an error occurs.

we put data in the weight variable,
Sometimes when I calculate the weight, it prints -0.0.
(When calculated in a negative state)

I'm not sure if I should report this issue, but I'll try to write it.

Perhaps it seems to judge that the values of 0 and -0.0 are the same. However, the actual value has changed, but inside livewire, the value has not changed, so I think it is an error that occurred.

```
Livewire\Exceptions\CorruptComponentPayloadException
Livewire encountered corrupt data when trying to hydrate the [sale-request] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.
```

### Stripped-down, copy-pastable code snippets

```php
  public $weight = 0;

  public function test()
  {
      $this->weight = -0.0;
  }

  public function test1()
  {
      return;
  }
```

```html
  <div class="mt-4 flex sm:mt-0 sm:ml-4">
      <button type="button" class="order-1 ml-3 inline-flex items-center px-4 py-2 border border-gray-300 shadow-sm text-sm font-medium rounded-md text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500 sm:order-0 sm:ml-0"
          wire:click="test()">
          Test
      </button>
      <button type="button" class="order-0 inline-flex items-center px-4 py-2 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-purple-600 hover:bg-purple-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-purple-500 sm:order-1 sm:ml-3"
          wire:click="test1()">
          Test1
      </button>
  </div>
```

**Context**
- Livewire version: 2.4.4
- Laravel version: 8.4.0
- Alpine version: 2.8.2
- Browser: Chrome

**Comment by nuernbergerA** [comment, marked as answer]
That's a known issue with JS

Try this in your console
```js
let i = 0.0
i // = 0
```

maybe we can do some smart casting if its a value that will change in js due to js casting we can do the same on the backend to have the same data

**Comment by ChrisSantiago82** [comment]
Is it the same than issue #2717 ?
If other people are having the same issue, maybe it's worth solving it?

**Comment by calebporzio** [comment]
Yeah, like @nuernbergerA said, we need to add a step in php to detect this case and force it to be a JS value that won't be changed between requests by the browser

**Comment by nuernbergerA** [comment]
For that reason I only work with int and never with floats

**Comment by ChrisSantiago82** [comment]
Yes, I think working with int would solve the problem. But I think it's difficult to tell everybody who uses livewire that they only should use int. 
But it's also interesting that PHP is fine with -0.0. Like it wouldn't just change it to 0 how Javascript and many other languages would do it.
